cmake/common/helpers_common.cmake

```diff
@@ -492,7 +495,17 @@ function(add_obs_plugin target)
     endforeach()
   endif()
 
-  if(found_platform)
+  if(_AOP_NUM_ARCHITECTURES EQUAL 0)
+    set(found_architecture TRUE)
+  else()
+    foreach(architecture IN LISTS _AOP_ARCHITECTURES)
+      if("${architecture}" STREQUAL CMAKE_SYSTEM_PROCESSOR)
```

Comment:
You cannot just check for `CMAKE_SYSTEM_PROCESSOR` because it will not match the build configuration on Windows or macOS (as neither uses this value for target architecture selection).

On macOS you'd need to check for `CMAKE_OSX_ARCHITECTURES` (which can be a list, i.e. `x86_64;arm64` because you can build universal binaries) and on Windows you need to check for `CMAKE_GENERATOR_PLATFORM` which can be either `Win32`, `x64`, or `ARM64`.

Reply:
Tell me if you are fine with the change/direction so I can squash.

Reply:
Please mark as resolved if you are okay with the change.

Reply:
See my other review regarding `CMAKE_OSX_ARCHITECTURES`.

Reply:
Done